### ISSUE TITLE
Allow NAT networks (like WSL2) to be Private instead of Public or Unidentified 

### ISSUE THREAD

**Issue by driver1998** [microsoft/WinDev#31]
Nowadays when you set up a NAT network on Windows (e.g.: Wire up to machines physically and use Internet Connection Sharing, or setting up NAT on Hyper-V/WSL2), the resulting network on the host side (where DHCP server is) is set as "unidentified".

Undefined means Public in Windows Firewall, so for example SMB sharing is blocked by default.

Sure I can open that up, but that means my SMB shares will also be open when I connect to public WiFi or something, no exactly secure.

In these NAT environments, I can control what machine can be connected (In the case of the WSL2 vEthernet, that will be only two, the host, and the WSL2 VM), so I would like to set these networks to be Private.

But I can't, because they are "Unidentified", even though they are defined on the machine itself.

**Comment by asklar** [comment]
Hi @driver1998, thanks for filing feedback! Since this is related to WSL, can you please post it in the WSL repo instead? I see there are a number of networking issues filed there already so maybe this is already covered by one of those :)
https://github.com/microsoft/wsl

Thanks!
